### PR TITLE
Added descriptive text to the links of the lessons and projects Bubbl…

### PR DESCRIPTION
**Pull request by Preslav977**
…e Sort, TicTacToe, OOP, and Files and Serialization.

## Because
I added more descriptive text to the links to the lessons and projects. Bubble Sort, OOP, TicTacToe, and Files and Serialization.

## This PR
This PR is provided in a couple of places in the lessons and projects more descriptive text with information.

## Issue

Related to #28290

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: ruby/object_oriented_programming_basics/object_oriented_programming.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ This section contains a general overview of topics that you will learn in this l
     - [Ruby Explained: Inheritance and Scope](http://www.eriktrautman.com/posts/ruby-explained-inheritance-and-scope)
 1. Read the article [Object Relationships in Basic Ruby](https://medium.com/@marcellamaki/object-relationships-in-basic-ruby-1af5773fff48) to see an example of how two classes can interact.
 1. Read the [Bastard's Chapter on Error Handling](http://ruby.bastardsbook.com/chapters/exception-handling/) to reinforce your understanding of dealing with errors.
-1. Do [Quiz #5](http://www.codequizzes.com/ruby/beginner/intro-object-oriented-programming) and [Quiz #7](http://www.codequizzes.com/ruby/beginner/modules-classes-inheritance) from Code Quizzes
+1. Do this [quiz on introductory object-oriented programming](http://www.codequizzes.com/ruby/beginner/intro-object-oriented-programming) and this [quiz on modules, classes, and inheritence](http://www.codequizzes.com/ruby/beginner/modules-classes-inheritance) from Code Quizzes.
```
Comment (MaoShizhong) [inline ruby/object_oriented_programming_basics/object_oriented_programming.md:29]: Oops, my typo.

```suggestion
1. Do this [quiz on introductory object-oriented programming](http://www.codequizzes.com/ruby/beginner/intro-object-oriented-programming) and this [quiz on modules, classes, and inheritance](http://www.codequizzes.com/ruby/beginner/modules-classes-inheritance) from Code Quizzes.
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Just the one typo fix below (which was my mistake).

`files_and_serialization.md` and `project_tic_tac_toe.md` both have some linting errors, if you're happy to sort them out? Instructions below if so - let me know if you're happy to apply them or not.

### `files_and_serialization.md`

1. In the "Knowledge check" section, put the following text between the heading and the list (**must** do this step first):
   ```text
   The following questions are an opportunity to reflect on key topics in this lesson. If you can't answer a question, click on it to review the material, but keep in mind you are not expected to memorize or master this knowledge.
   ```
2. Following out [local linting and fixing instructions](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md#curriculum-linting), run the following command **twice**, which should fix all the lint errors remaining:
   ```bash
   npm run fix:lesson -- "ruby/files_and_serialization/files_and_serialization.md"
   ```

### `project_tic_tac_toe.md`

1. Run the following command twice (make sure to use the project script, not the lesson script like before):
   ```bash
   npm run fix:project -- "ruby/object_oriented_programming_basics/project_tic_tac_toe.md"
   ```

File: ruby/object_oriented_programming_basics/object_oriented_programming.md (hunk 1 above)
Comment (Preslav977) [reply ruby/object_oriented_programming_basics/object_oriented_programming.md:29]: Okay. Will do.

Comment (MaoShizhong) [reply ruby/object_oriented_programming_basics/object_oriented_programming.md:29]: In addition to the typo fix above, could you also make the following quick change?

### `object_oriented_programming.md`

1. Remove line 52
2. Unindent the next 2 lines (so they're part of the main list and not a nested list)
3. Amend the "99 Bottles of OOP"'s last sentence to say
   ```text
   This is a great resource if, after completing several OOP projects, you still feel uncertain about OOP concepts and would like to be guided through some refactoring examples and explanations.
   ```

Comment (Preslav977) [reply ruby/object_oriented_programming_basics/object_oriented_programming.md:29]: Okay. No problem.

Comment (Preslav977) [reply ruby/object_oriented_programming_basics/object_oriented_programming.md:29]: Quick question: after I made the changes, should I run the linter the same as I did for the lesson, obviously on a different path, or not?

Comment (MaoShizhong) [reply ruby/object_oriented_programming_basics/object_oriented_programming.md:29]: Run the `lint:lesson` script with the appropriate file path to see if there are still any lint errors. If 0, you're good to push. If any remain, read the errors and fix accordingly. Most errors are fixable with the appropriate `fix:*` script, but some errors won't be fixable by it and need to be manually fixed based on context.

File: ruby/object_oriented_programming_basics/object_oriented_programming.md [hunk 2]
```diff
@@ -49,6 +49,5 @@ This section contains helpful links to related content. It isn't required, so co
...
-- After you have completed several OOP projects, the following books by Sandi Metz are essential to build on the foundational concepts in this lesson.
-  - [99 Bottles of OOP](https://sandimetz.com/99bottles) is a hands-on workbook that you should code along with to get the most out of it. This is a great resource if you feel uncertain about OOP concepts and would like to guided through refactoring examples and explanations.
-  - [Practical Object-Oriented Design in Ruby](https://www.poodr.com/) is a traditional technical book with practical examples. This is a great resource after 99 Bottles or if you already feel like you fully understand OOP concepts.
+- [99 Bottles of OOP](https://sandimetz.com/99bottles) is a hands-on workbook that you should code along with to get the most out of it. This is a great resource if you feel uncertain about OOP concepts and would like to guided through refactoring examples and explanations.
```
Comment (MaoShizhong) [inline ruby/object_oriented_programming_basics/object_oriented_programming.md:52]: As per step 3 of https://github.com/TheOdinProject/curriculum/pull/28362/files#r1668589269
```suggestion
- [99 Bottles of OOP](https://sandimetz.com/99bottles) is a hands-on workbook that you should code along with to get the most out of it. This is a great resource if, after completing several OOP projects, you still feel uncertain about OOP concepts and would like to be guided through some refactoring examples and explanations.
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Could you also fix the "inheritance" typo highlighted in the previous review comment? Many thanks - should be good to merge once these two things are changed.

File: ruby/object_oriented_programming_basics/object_oriented_programming.md (hunk 2 above)
Comment (Preslav977) [reply ruby/object_oriented_programming_basics/object_oriented_programming.md:52]: Oh, I didn't see those, sorry.

**Review by MaoShizhong — APPROVED** [review]
Thank you for contributing, and congrats on your first PR merged here! :rocket:

**Pull request merged**